Add SQL Wildcards, Update LIKE and SELECT (#585)
Codecademy/docs: content/sql/concepts/wildcards/wildcards.md

---

> Title: Wildcards

Reviewer: ```suggestion
Title: 'Wildcards'
```

Description: 'Wildcards are special characters used in SQL to represent one or more arbitrary characters.'
Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Operators'
  - 'Database'
  - 'Queries'
  - 'SQLite'
  - 'MySQL'
  - 'PostgreSQL'
CatalogContent:
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
  - 'paths/design-databases-with-postgresql'
---

Wildcards are special characters used in SQL to represent one or more arbitrary characters.

There are two wildcards generally recognized across SQL implementations:

- `%` which matches zero to any arbitrary number of characters.
- `_` which matches any one single arbitrary character.

> Some SQL implementations add additional types of wildcards. (Microsoft Access SQL, in particular, is non-standard, using `*` in place of `%` and `?` in place of `_`.)

Reviewer: I don't think this sentence needs to be in parenthesis:

```suggestion
Some SQL implementations add additional types of wildcards. Microsoft Access SQL, in particular, is non-standard, using `*` in place of `%` and `?` in place of `_`.
```

Wildcards are typically used with the [`LIKE`](https://www.codecademy.com/resources/docs/sql/operators/like) operator, which returns true if the first string argument matches the wildcard pattern in the second.

## Examples

> Select all rows where `column`'s value has "H" as the second character:

Reviewer 2: ```suggestion
The following example selects all rows where "H" is the second character in the value of `column`:
```

I'm personally curious, what is the `%` doing in this example? It's checking for zero or more instances of such a `column` value?

Author: It's matching the rest of the string after the 'H'. Without it, it would only match a two character string ending with 'H'.

Reviewer 2: Oh ok gotcha. 🙏

```sql
SELECT * FROM table WHERE column LIKE '_H%';
```

The matches would include values like "THE" and "WHERE" but not "HOUSE", or "BREATH".

> Select all rows where `column`'s value contanes an "H" anywhere:

Reviewer: ```suggestion
Select all rows where `column`'s value contains an "H" anywhere:
```

```sql
SELECT * FROM table WHERE column LIKE '%H%';
```

> Matches "THE", "WHERE", "HOUSE" and "BREATH".

Reviewer 2: ```suggestion
The matches would include any pattern with an "H", including "THE", "WHERE", "HOUSE", and "BREATH".
```

Select all rows where `column`'s value is three characters long and has an "H" as the second character:

```sql
SELECT * FROM table WHERE column LIKE '_H_';
```

> Matches the value "THE" but not "WHERE", "HOUSE" or "BREATH".

Reviewer: I think these wildcard examples do well in this entry on wildcards, but maybe not so well in the `LIKE` operator entry. Can we swap out those duplicated examples in the `LIKE` entry with maybe just one example?

When the wildcard link is brought up in the `LIKE` entry, maybe we can also refer to it so the reader is prompted to go there for more examples.

Reviewer: Thanks for incorporating this @SSwiniarski!

> Matches the value "THE" but not "WHERE", "HOUSE" or "BREATH".

Reviewer 3: Would it also be helpful to include an example of begins with/ ends with? e.g. H% or %H

Author: Good idea.

> Matches "THE" but not "WHERE", "HOUSE" or "BREATH".

Reviewer 2: ```suggestion
The matches would include patterns like "THE" but not "WHERE", "HOUSE", or "BREATH".
```
